src/tribler/core/components/libtorrent/tests/test_download_manager.py

```diff
@@ -455,3 +492,79 @@ async def test_check_for_dht_ready(fake_dlmgr):
     fake_dlmgr.get_session().status().dht_nodes = 1000
     # If the session has enough peers, it should finish instantly
     await fake_dlmgr._check_dht_ready()
+
+
+def test_update_trackers(fake_dlmgr: DownloadManager) -> None:
+    fake_download, dl_state = create_fake_download_and_state()
+    fake_dlmgr.downloads[fake_download.infohash] = fake_download
+    fake_metainfo = {b'info': {b'name': b"test_download"}}
+    fake_download.get_def().metainfo = fake_metainfo
+
+    fake_tracker1 = "127.0.0.1/test-announce1"
+
+    fake_dlmgr.update_trackers(fake_download.infohash, [fake_tracker1])
+
+    assert fake_metainfo["announce"] == fake_tracker1
+    assert "announce-list" not in fake_metainfo
+
+
+def test_update_trackers_list(fake_dlmgr: DownloadManager) -> None:
+    fake_download, dl_state = create_fake_download_and_state()
+    fake_dlmgr.downloads[fake_download.infohash] = fake_download
+    fake_metainfo = {b'info': {b'name': b"test_download"}}
+    fake_download.get_def().metainfo = fake_metainfo
+
+    fake_tracker1 = "127.0.0.1/test-announce1"
+    fake_tracker2 = "127.0.0.1/test-announce2"
+
+    fake_dlmgr.update_trackers(fake_download.infohash, [fake_tracker1, fake_tracker2])
+
+    assert "announce" not in fake_metainfo
+    # The order of the results changes between OSes, so this is needed
+    assert (
+        fake_metainfo["announce-list"] == [[fake_tracker1, fake_tracker2]] or
+        fake_metainfo["announce-list"] == [[fake_tracker2, fake_tracker1]]
+    )
```

Comment:
Using the approach with the set looks a bit more correct and generic:

```suggestion
    tracker_list = fake_metainfo["announce-list"]
    actual_trackers = set(itertools.chain.from_iterable(tracker_list))
    assert actual_trackers == {fake_tracker1, fake_tracker2}
```